src/battle_ai_util.c

```diff
@@ -2328,6 +2328,19 @@ bool32 IsSwitchOutEffect(u32 effect)
     }
 }
 
+bool32 IsSleepClauseTriggeringEffect(u32 effect)
```

Comment:
can this be a more generic name? The function name and the comment down under imply that it's contents have something to do with sleep clause which is not the case because the content is just a generic sleep effect check.

Reply:
I don't mind either way as long as it is consistent. Either make it generic or include all the sleep clause check there

Reply:
renamed to `IsEffectBlockedBySleepClause` and moved the sleep clause check inside it